lib/vacuum/prune/pack/pack.go

```diff
@@ -96,47 +96,37 @@ type Application struct {
 type packageService interface {
 	GetRepositories() ([]string, error)
 	GetPackages(respository string) ([]pack.PackageEnvelope, error)
+	ReadPackageEnvelope(loc.Locator) (*pack.PackageEnvelope, error)
 	DeletePackage(loc.Locator) error
 }
 
-// Prune removes unused packages from the conigured package service.
+// Prune removes unused packages from the configured package service.
 // It uses the direct application dependencies to determine the set of packages
 // that are still required, and sweeps the rest.
 // It will not remove packages from repositories other than the defaults.SystemAccountOrg
 // unless it can tell if a package is safe to remove.
 func (r *cleanup) Prune(context.Context) error {
-	repositories, err := r.Packages.GetRepositories()
+	required, err := r.mark()
 	if err != nil {
 		return trace.Wrap(err)
 	}
 
-	required, err := r.mark()
+	state, err := r.build(required)
```

Comment:
I just got a thought: cleanup is an upgrade phase now, right? Is it enabled for Ops Center app too? When Ops Center is upgraded, won't it nuke all apps in it? Maybe the gc command should just not work for Ops Centers too.

Reply:
Upgrade only cleans up obsolete `journald` directories.
`gravity gc` will not touch additional applications (and their dependencies) as it works off of a set of required dependencies based on the cluster application.
So it would only prune older versions of all packages directly or indirectly related to the cluster application and leave the rest.

Reply:
Oh, yes, that it will.

Reply:
Got it, if upgrade doesn't cleanup much else beside logs, it should be fine.

The fact that it doesn't make much sense doesn't mean nobody will try it :) Honestly, I'm not sure what kind of behavior would be best when you run it on Ops Center. I would think that we should not implicitly remove any apps from it. In any case, that's out of the scope of this PR.